README.md

```diff
@@ -192,205 +141,129 @@ data SwapDatum
+### Benchmarks and Fee Estimations (YMMV)
+Thanks to the efficiency of Aiken, Cardano-Swaps is capable of composing up to 14 different swaps in a single transaction, with a total transaction fee of 1.5 ADA. A transaction with a single swap only costs 0.25 ADA. These results likely make this the cheapest "DEX" on Cardano by an order of magnitude.
 
-:pushpin: v1.0.0 of cardano-swaps used unique spending scripts; you can read about the limitations in the v1.0.0 commit README.
+Given the performance of these Aiken contracts, Cardano-Swaps is more than performant enough for the current state of Cardano. **No CIPs or hard-forks are needed. This protocol works on the Cardano blockchain, as is.**
 
-#### What if two arbitragers compete for the same swap?
-Recall the contrived example above. What would happen if Charlie and Mike try to arbitrage it at the same time?
+Full benchmarking details can be found in the [Benchmarks](Benchmarks.md) file.
 
-1. Charlie and Mike both build their transactions since the UTxOs still exist.
-2. Charlie and Mike submit their transaction at the same time.
-3. Charlie's is added to a block first.
-4. When Mike's transaction is then picked to go into a block, the UTxOs no longer exist. The transaction fails without needing to run the swap contracts.
 
-Since Mike's transaction will fail without needing to run the swap script, Mike's collateral is safe. Further, the more available swaps there are, the less likely these "collisions" will occur.
+## Future Directions
+Features and considerations for future versions of Cardano-Swaps:
 
-#### If Cardano-Swaps reaches mass adoption, won't TVL on Cardano go down?
+### Offer-Based Swap Addresses
+Currently, swap addresses are unique for every trading pair. This necessitates many distinct Beacon UTxOs (and associated deposits), even if users are offering the same asset in exchange for varying assets. Future versions of Cardano-Swaps will instead utilize "Offer-based" swap addresses, which are unique only to the asset being offered. The desired "exchange" asset will be specified in the datum of every "Swap" UTxO at the address. This will dramatically lower Beacon UTxO deposit fees, as only one such UTxO will be needed per "Offer" asset, per user. This will also improve querying efficiency; instead of having to lookup a specific trading pair, users can lookup all swap addresses offering Token "ABC", and then filter through the desired exchange asset(s).
```

Comment:
This is not entirely correct. There will still be the trading pair specific beacon that will go with every swappable utxo. Also the motivation for this is strictly arbitrage querying. The deposit is not a big deal imo. It would be nice to have an additional filtered query based on asset being offered. Basically Swaps would support two queries:
1. Show me all swaps for AGIX/ADA (this is useful for normal users)
2. Show me all swaps offering ADA (this is useful for arbitragers)

Also the tone of this should reflect that this is just a suggestion. Idk if it will work the way I am imagining but it is worth trying. The way you have it worded makes it sounds like this is obviously better when the trade-offs are not fully known yet. It is just an avenue to explore.

Reply:
The offer beacon would be the address level beacon while every swap would have the trading pair beacon.

Reply:
the deposit is definitely a big deal considering its set at 20 ADA. thats hundreds of extra ADA, per swap pair, per user.